comid/integrityregisters.go

```diff
@@ -136,3 +137,32 @@ func (i *IntegrityRegisters) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func (i IntegrityRegisters) Equal(r IntegrityRegisters) bool {
+	return reflect.DeepEqual(i, r)
+}
+
+// CompareAgainstReference checks if IntegrityRegisters object matches with a reference
+//
+//	See the following CoRIM spec for rules to compare
+//	IntegrityRegisters against a reference:
+//	https://ietf-rats-wg.github.io/draft-ietf-rats-corim/draft-ietf-rats-corim.html#name-comparison-for-integrity-re
+func (i IntegrityRegisters) CompareAgainstReference(r IntegrityRegisters) bool {
+	result := false
+
+	for refIndex, refDigests := range r.IndexMap {
+		claimDigests, ok := i.IndexMap[refIndex]
+		if !ok {
+			return false
+		}
+
+		ref := refDigests
```

Comment:
Is ref needed ?

Reply:
I got a linter error if I used refDigests directly. So, I created a copy of it.